### PR TITLE
:sparkles: Enable consumers to request service providers by priority …

### PR DESCRIPTION
**Pull request by tonycody**
Related issue: #10574

:sparkles: Enable consumers to request service providers by priority using the appropriate serialization type

Such as:
prefer_serialization > serialization > default serialization

fastjson2,hessian2 > fastjson > hessian2

Use the order: fastjson2 > hessian2 > fastjson

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/ProviderConstants.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package org.apache.dubbo.common.constants;
+
+/**
+ * Provider Constants
+ *
+ * @author XS <tonycody@qq.com>
+ * @date: 2022/9/1 16:59:53
+ * @since 1.0
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/constants/ProviderConstants.java:8]: remove these tags pls

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/ProviderConstants.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.apache.dubbo.common.constants;
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/constants/ProviderConstants.java:1]: Please add ASF license header to all files

File: dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java [hunk 3]
```diff
@@ -110,13 +110,7 @@
...
 import static org.apache.dubbo.registry.Constants.REGISTRY_RETRY_PERIOD_KEY;
 import static org.apache.dubbo.registry.Constants.SIMPLIFIED_KEY;
-import static org.apache.dubbo.remoting.Constants.BIND_IP_KEY;
-import static org.apache.dubbo.remoting.Constants.BIND_PORT_KEY;
-import static org.apache.dubbo.remoting.Constants.CHECK_KEY;
-import static org.apache.dubbo.remoting.Constants.CODEC_KEY;
-import static org.apache.dubbo.remoting.Constants.CONNECTIONS_KEY;
-import static org.apache.dubbo.remoting.Constants.EXCHANGER_KEY;
-import static org.apache.dubbo.remoting.Constants.SERIALIZATION_KEY;
+import static org.apache.dubbo.remoting.Constants.*;
```
Comment (AlbumenJ) [inline dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryProtocol.java:113]: do not use comma import

File: dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvokerAvailableTest.java [hunk 4]
```diff
@@ -151,6 +151,26 @@ public void test_Lazy_ChannelReadOnly() throws Exception {
         Assertions.assertFalse(invoker.isAvailable());
     }
 
+    /**
+     * The test prefer serialization
+     *
+     * @throws Exception 异常
```
Comment (AlbumenJ) [inline dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/DubboInvokerAvailableTest.java:157]: comment in eng

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by tonycody** [comment]
ping

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/builders/AbstractServiceBuilder.java [hunk 5]
```diff
@@ -211,11 +216,22 @@ public B warmup(Integer warmup) {
...
-    public  B serialization(String serialization) {
+    public B serialization(String serialization) {
         this.serialization = serialization;
         return getThis();
     }
 
+    /**
+     * The prefer serialization type
+     *
+     * @param preferSerialization 喜欢序列化
```
Comment (AlbumenJ) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/bootstrap/builders/AbstractServiceBuilder.java:227]: comment in eng

File: dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/utils/UrlUtils.java [hunk 6]
```diff
@@ -18,7 +18,15 @@
 package org.apache.dubbo.remoting.utils;
 
 import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.serialize.support.DefaultSerializationSelector;
+import org.apache.dubbo.common.utils.StringUtils;
 import org.apache.dubbo.remoting.Constants;
+import org.apache.dubbo.remoting.transport.CodecSupport;
+
+import java.util.*;
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/utils/UrlUtils.java:26]: do not use comma import

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**